pulp_rpm/app/access_policy.py

```diff
@@ -20,20 +20,21 @@ def has_perms_to_copy(self, request, view, action):
         serializer = view.serializer_class(data=request.data, context={"request": request})
         serializer.is_valid(raise_exception=True)
 
-        dest_repo_id = request.data["config"][0]["dest_repo"].split("/")[-2]
-        dest_repo = RpmRepository.objects.get(pk=dest_repo_id)
+        for copy_action in serializer.data["config"]:
+            dest_repo_id = copy_action["dest_repo"].split("/")[-2]
+            dest_repo = RpmRepository.objects.get(pk=dest_repo_id)
 
-        source_repo_id = request.data["config"][0]["source_repo_version"].split("/")[-4]
-        source_repo_ver_number = request.data["config"][0]["source_repo_version"].split("/")[-2]
+            source_repo_id = copy_action["source_repo_version"].split("/")[-4]
+            source_repo_ver_number = copy_action["source_repo_version"].split("/")[-2]
 
-        source_repo_version = RepositoryVersion.objects.get(
-            repository_id=source_repo_id, number=source_repo_ver_number
-        )
-        source_repo = RpmRepository.objects.get(pk=source_repo_version.repository_id)
+            source_repo_version = RepositoryVersion.objects.get(
+                repository_id=source_repo_id, number=source_repo_ver_number
+            )
+            source_repo = RpmRepository.objects.get(pk=source_repo_version.repository_id)
 
-        if request.user.has_perm("rpm.change_rpmrepository", dest_repo) and request.user.has_perm(
-            "rpm.view_rpmrepository", source_repo
-        ):
-            return True
+            if not request.user.has_perm(
+                "rpm.modify_content_rpmrepository", dest_repo
+            ) and not request.user.has_perm("rpm.view_rpmrepository", source_repo):
+                return False
 
-        return False
+        return True
```

Comment:
I believe you still should add check for model level permissions. Right now you're checking whether user has permission on a specific object, he could also have model-level permission which would span across *all* instances.

Reply:
In addition, I think current logic is faulty - if user does not have  modify permission on dest repo but does not have view permission on source repo? True will be returned :)

Reply:
I might be missing something but I'm not sure that happens, it looks like it would just `return False` early when it hits any repo for which it doesn't have permissions (but maybe this was addressed prior to my comment)